lazer/sdk/js/src/util/env-util.ts

```diff
@@ -0,0 +1,34 @@
+// we create this local-only type, which has assertions made to indicate
+// that we do not know and cannot guarantee which JS environment we are in
+const g = globalThis as Partial<{
+  self: typeof globalThis.self;
+  window: typeof globalThis.window;
+}>;
+
+/**
+ * Detects if this code is running within any Service or WebWorker context.
+ * @returns true if in a worker of some kind, false if otherwise
+ */
+export function envIsServiceOrWebWorker() {
+  const possiblyInAWorker =
+    typeof WorkerGlobalScope !== "undefined" && g.self !== undefined;
+  return possiblyInAWorker && g.self instanceof WorkerGlobalScope;
```

Comment:
Can this be minorly more concisely written as:

```ts
return WorkerGlobalScope !== "undefined" && g.self instanceof WorkerGlobalScope;
```

Isn't the check for `g.self !== undefined` redundant?

Reply:
done in 8baec64433940479ac9079f537be0c782207f205